core/document.ts

```diff
@@ -39,11 +33,7 @@ export default class Document {
   }
 
   async read() {
-    if (this.#data === undefined) {
-      this.#data = await this.#entry.readData();
-    }
```

Comment:
This is when it will fail on creating new documents. the `isNew`. It should be something like:

```js
if (this.isNew) {
  return {};
}
return await this.#entry.readData();
```